fvm/evm/types/emulator.go

```diff
@@ -14,12 +15,20 @@ var (
 	BlockNumberForEVMRules = big.NewInt(1)
 )
 
+type Precompile interface {
+	gethVM.PrecompiledContract
+	Address() Address
+}
+
 // BlockContext holds the context needed for the emulator operations
 type BlockContext struct {
 	BlockNumber            uint64
 	DirectCallBaseGasUsage uint64
 	DirectCallGasPrice     uint64
 	GasFeeCollector        Address
+
+	// a set of extra precompiles to be injected
+	ExtraPrecompiles []Precompile
```

Comment:
`FlowPrecompiles`?

Reply:
I'm trying to keep the emulator not referencing any Flow protocol concepts, we might even add some precompiles in the future that is not Flow-related.